List/Foldable

```diff
@@ -0,0 +1,6 @@
+    let Foldable = ../Foldable/Type
+
+in  let fold =
+          https://ipfs.io/ipfs/QmQ8w5PLcsNz56dMvRtq54vbuPe9cNnCCUXAQp6xLc6Ccx/Prelude/List/fold
```

Comment:
I have mixed feelings about using a url reference in a core library (is it fair to predict that kind of status for dhall's Foldable?). If this is a temporary convenience, then of course I have no objection. But in the long term, are there ideas about how we could avoid the baked-in network cost? Maybe if `Foldable` gets enough use, it could be added to the Prelude, so that the whole collection could be downloaded together and connected with relative filepaths?

Reply:
Afaik I can't interact with `List` without using the functions `Gabriel` provides. There's no such thing as pattern matching on `:` or `cons`. 

Two options: 
* Hosting `Prelude` locally in `dhall-bhat` and basically making this a `ClassyPrelude` :P 
* @sellout has been bouncing around the idea of a `dhall-preprocess` for prefetching URLs into 
a `build` folder and referencing that instead.

I think point 2 is better in that it solves more cases. Point 1 can be undertaken in the meantime but obviously I'd prefer do a separate Issue/PR for it.

Reply:
Yeah, we don’t really need a separate app, I realized – we can use something like `dhall --pretty <<< file >preprocessed` as a build tool to resolve everything.

But `List/fold` doesn’t require a URL, see https://github.com/sellout/dada/blob/master/List/recursive#L8